tls/s2n_config.h

```diff
@@ -69,8 +69,7 @@ struct s2n_config {
     unsigned disable_x509_time_validation : 1;
     unsigned disable_x509_validation : 1;
     unsigned max_verify_cert_chain_depth_set : 1;
-    /* Whether to add dss cert type during a server certificate request.
```

Comment:
this one doesn't have a good destination in the new usage guide?

Reply:
Yeah, I can't remember our usage guide _ever_ mentioning anything about DSS cert types. I'd rather just remove the link for now.

Reply:
Maybe it dates back to when the usage guide was mostly a list of APIs? That field exists entirely for this sketchy API: https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L3149-L3155